### DISCUSSION TITLE
Floating point exception

### DISCUSSION THREAD

**Discussion by leishan** [nodejs/help#4770] [Q&A]
### Version

v18.16.0

### Platform

MIPS64 GNU/Linux

### What do you see instead?

I have a machine with a MIPS64 CPU architecture, and since node-v18.16.0 does not have a corresponding binary installer file for it, I compiled the source code on this machine using `node-v18.16.0.tar.xz`.

My first step is installing `libssl-dev`

The second step is to modify the `Makefile` file at line 937 of the

```diff
else
-DESTCPU ? = x86
+DESTCPU ? = mips64
```

and changed the `Makefile` file at line 975 of the

```diff
else
-ARCH=x86
+ARCH=mips64
```

Finally, compile the node-v18.16.0 source code using the following command:

```shell
./configure --prefix=/usr/local/mynode
make
make install
```

This compilation went well and the generated `node` commands worked fine, but using commands such as `npm install xxx` and `npm ping` prompted the following error:

```console
node: malloc.c:2385: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old _top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.

Abandoned
```

So, I downloaded the v16.20.0 source code and compiled it again, the compilation went well and the generated `node` commands worked fine, but commands like `npm install xxx` and `npm ping` prompted the following error:

```console
floating point exception
```

Later, I tried to use the node-v18.16.0 and node-v16.20.0 generated node commands to test the following two files `a.js` and `b.js` respectively, and found that `node a.js` works normally, but `node b.js` will have the above "abandoned" and "floating point exception" respectively under node commands. The error message "floating point exception" appears in `b.js` under node command.

this is `a.js`:

```js
const http = require('http')
const options = {
  hostname: '192.168.1.2',  
  port: 80,
  path: '/',
  method: 'GET'
}

const req = http.request(options, res => {
  console.log(`statusCode: ${res.statusCode}`)

  res.on('data', d => {
    console.log(d.toString())
  })
})

req.on('error', error => {
  console.error(error)
})

req.end()
```

this is `b.js`:

```js
const https = require('https')
const options = {
  hostname: 'www.github.com',  
  port: 443,
  path: '/',
  method: 'GET',
//rejectUnauthorized:false
}

const req = https.request(options, res => {
  console.log(`statusCode: ${res.statusCode}`)

  res.on('data', d => {
    console.log(d.toString())
  })
})

req.on('error', error => {
  console.error(error)
})

req.end()
```

Later, I downloaded the `node-v18.16.0-linux-x64.tar.xz` binary package and installed it on both x64 and Arm64 machines, and there was no "abandoned" or "floating point exception " problems.

I used `ldd` and `readelf` commands to compare the `node` commands generated by the `node-v18.16.0.tar.xz` compilation on the MIPS64 machine with the `node-v18.16.0-linux-x64.tar.xz` binary installation on the x86/Arm64 machine, and found that there is not much difference between the two.

What is the problem? I've been puzzled.

**Comment by targos** [comment]
@kapouer might be able to help you, but note that there is no official support for mips64.

**Comment by kapouer** [comment]
Hi, indeed, we have a working nodejs 18 on debian/bookworm for mips64el. I suppose you are not using that distribution.
A good place to find cues on how to build it for mips (which flags, which patches) is:
https://salsa.debian.org/js-team/nodejs/-/blob/master-18.x/debian/rules
and https://salsa.debian.org/js-team/nodejs/-/blob/master-18.x/debian/patches
First thing is to check what is your CPU model...

**Comment by leedejun** [comment]
Hello, @leishan 
Is your problem solved? I also have the same problem as you.
